Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -4667,6 +4688,488 @@ def apply_quarantine_file_action_command(args: dict) -> CommandResults:
+                                   for host in resource.get('metadata', []) if (filecount := host.get('filecount', {}))),
+        'Description': resource.get('description'),
+        'Hosts/Host groups': resource.get('hosts') or resource.get('host_groups'),
+        'Start time': resource.get('scan_started_on'),
+        'End time': resource.get('scan_completed_on'),
+        'Run by': resource.get('created_by')
+    }
+    return output
+
+
+def ODS_get_scan_resources_to_human_readable(resources: list[dict]) -> str:
+
+    human_readable = tableToMarkdown(
+        'CrowdStrike Falcon ODS Scans',
+        [map_scan_resource_to_UI(resource) for resource in resources],
+        headers=['ID', 'Status', 'Severity', 'File Count', 'Description',
+                 'Hosts/Host groups', 'End time', 'Start time', 'Run by']
+    )
+
+    return human_readable
+
+
+def get_ODS_scan_ids(args: dict) -> list[str] | None:
+
+    demisto.debug('Fetching IDs from query api')
+
+    query_filter = build_cs_falcon_filter(
+        custom_filter=args.get('filter'),
+        initiated_from=args.get('initiated_from'),
+        status=args.get('status'),
+        severity=args.get('severity'),
+        scan_started_on=args.get('scan_started_on'),
+        scan_completed_on=args.get('scan_completed_on'),
+    )
+
+    raw_response = ODS_query_scans_request(
+        filter=query_filter,
+        offset=args.get('offset'),
+        limit=args.get('limit'),
+    )
+
+    return raw_response.get('resources')
+
+
+@polling_function(
+    'cs-falcon-ods-query-scan',
+    polling_arg_name='wait_for_result',
+    interval=arg_to_number(dict_safe_get(demisto.args(), ['interval_in_seconds'], 0, (int, str))),
+    timeout=arg_to_number(dict_safe_get(demisto.args(), ['timeout_in_seconds'], 0, (int, str))),
+)
+def cs_falcon_ODS_query_scans_command(args: dict) -> PollResult:
+    # call the query api if no ids given
+    ids = argToList(args.get('ids')) or get_ODS_scan_ids(args)
+
+    if not ids:
+        command_results = CommandResults(readable_output='No scans match the arguments/filter.')
+
+    else:
+        response = ODS_get_scans_by_id_request(ids)
+        resources = response.get('resources', [])
+        human_readable = ODS_get_scan_resources_to_human_readable(resources)
+
+        command_results = CommandResults(
+            raw_response=response,
+            outputs_prefix='CrowdStrike.ODSScan',
+            outputs_key_field='id',
+            outputs=resources,
+            readable_output=human_readable,
+        )
+
+    scan_in_progress = any(dict_safe_get(scan, ('status',), return_type=str) in ('pending', 'running')
+                           for scan in command_results.outputs)  # type: ignore[attr-defined]
+
+    return PollResult(response=command_results,
+                      continue_to_poll=scan_in_progress,
+                      partial_result=command_results,
```

Comment:
if returning a partial result make sure to pop the ids we already returned so we dont return their results again

Reply:
it does, i checked